ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -75,9 +75,22 @@ export default Route.extend({
 
   beforeModel({ to: { queryParams } }) {
     const secret = this.secretParam();
+    const secretEngine = this.modelFor('vault.cluster.secrets.backend');
     return this.buildModel(secret, queryParams).then(() => {
       const parentKey = parentKeyForKey(secret);
       const mode = this.routeName.split('.').pop();
+      // for kv v2, redirect users from the old url to the new engine url (1.15.0 +)
+      if (secretEngine.type === 'kv' && secretEngine.version === 2) {
+        // if no secret param redirect to the create route
+        // if secret param they are either viewing or editing secret so navigate to the details route
+        return !secret
+          ? this.router.transitionTo('vault.cluster.secrets.backend.kv.create', secretEngine.id)
+          : this.router.transitionTo(
```

Comment:
because I didn't want to deal with the kv secret versions here and metadata permissions. So I thought I'd land on details and they could navigate from there. Do you think that's acceptable?

Reply:
I guess I'm confused that we know the `mode` so couldn't we use that to determine whether to send them to details vs edit? I'm not sure I follow why the version and permissions would be relevant here? Since those are handled by those respective routes

Reply:
Discussed with Claire the issues encountered here. Determined okay to keep as is.